## Comparing Average Bleu Scores

### For original data

In [ ]:
!pip install torch
!pip install tqdm
!pip install nltk
!pip install numpy
!pip install scikit-learn

In [ ]:
!python lmg_cc2ftr.py -train -train_data train.pkl -dictionary_data dict.pkl -batch_size 32

100%|█████████████████████████████████████████| 691/691 [06:46<00:00,  1.70it/s]
Training: Epoch 1 / 2 -- Total loss: 545.061462
100%|█████████████████████████████████████████| 691/691 [06:44<00:00,  1.71it/s]
Training: Epoch 2 / 2 -- Total loss: 480.595276
--------------------------------------------------------------------------------
--------------------------Finish the training process---------------------------
--------------------------------------------------------------------------------


In [ ]:
!python lmg_cc2ftr.py -predict -pred_data test.pkl -dictionary_data dict.pkl -load_model cc2ftr.pt -name output.pkl

100%|█████████████████████████████████████████| 277/277 [00:20<00:00, 13.53it/s]
--------------------------------------------------------------------------------
--------------------------Finish the extracting process-------------------------
--------------------------------------------------------------------------------


In [ ]:
!python lmg_eval.py -train_data train.pkl -test_data test.pkl -train_cc2ftr_data train_cc2ftr.pkl -test_cc2ftr_data test_cc2ftr.pkl 

100%|█████████████████████████████████████| 2216/2216 [3:07:04<00:00,  5.07s/it]
Average of blue scores: 20.579069529524507


#### Average Bleu Scores of original data: 20.579069529524507

### For new data

In [ ]:
!pip install torch
!pip install tqdm
!pip install nltk
!pip install numpy
!pip install scikit-learn

In [ ]:
!python lmg_cc2ftr.py -train -train_data newtrain.pkl -dictionary_data newdict.pkl -batch_size 32

100%|█████████████████████████████████████████| 691/691 [06:49<00:00,  1.69it/s]
Training: Epoch 1 / 2 -- Total loss: 542.091370
100%|█████████████████████████████████████████| 691/691 [06:44<00:00,  1.71it/s]
Training: Epoch 2 / 2 -- Total loss: 481.489777
--------------------------------------------------------------------------------
--------------------------Finish the training process---------------------------
--------------------------------------------------------------------------------


In [ ]:
!python lmg_cc2ftr.py -predict -pred_data newtest.pkl -dictionary_data newdict.pkl -load_model snapshot/2021-12-20_08-02-07/epoch_2.pt -name newoutputtest.pkl

100%|█████████████████████████████████████████| 277/277 [00:20<00:00, 13.54it/s]
--------------------------------------------------------------------------------
--------------------------Finish the extracting process-------------------------
--------------------------------------------------------------------------------


In [ ]:
!python lmg_eval.py -train_data newtrain.pkl -test_data newtest.pkl -train_cc2ftr_data train_cc2ftr.pkl -test_cc2ftr_data test_cc2ftr.pkl 

100%|█████████████████████████████████████| 2216/2216 [3:14:15<00:00,  5.26s/it]
Average of blue scores: 21.113468382170318


#### Average Bleu Scores of new data: 21.113468382170318